File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();

        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0)+1);
        }

        List<Integer> uniqueList = new ArrayList<>(map.keySet());
        uniqueList.sort((a, b) -> map.get(b).compareTo(map.get(a)));

        int[] result = new int[k];
        for(int i=0; i<k; i++){
            result[i] = uniqueList.get(i);
        }
        return result;
    }
}

File: 347-top-k-frequent-elements/top-k-frequent-elements.dart
class Solution {
  List<int> topKFrequent(List<int> nums, int k) {
    Map<int, int> map = {};

    for(int i=0; i<nums.length; i++){
        int n = nums[i];
        map[n] = (map[n] ?? 0)+1;
    }
    List<int> uniqueList = map.keys.toList();
    uniqueList.sort((a, b) => map[b]!.compareTo(map[a]!));
    return uniqueList.take(k).toList();
  }
}